draftjs_exporter/dom_engine.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import absolute_import, unicode_literals
+
+import re
+
+# Python 2/3 unicode compatibility hack.
+# See http://stackoverflow.com/questions/6812031/how-to-make-unicode-string-with-python3
+try:
+    UNICODE_EXISTS = bool(type(unicode))
+except NameError:
+    def unicode(s):
+        return str(s)
+
+# BeautifulSoup import and helpers.
+try:
+    from bs4 import BeautifulSoup
```

Comment:
Ideally, each engine would live in its own module, so we would have something like this:
```
-- draftjs_exporter/
   |-- engines/
   |   |-- __init__.py       # Empty
   |   |-- base.py          # With the DOMEngine definition
   |   |-- html5lib.py     # With the DOM_HTML5LIB definition
   |   `-- lxml.py           # With the DOM_LXML definition
   `-- ...
```
And within each engine module, the import would be done normally (i.e. without wrapping them inside a try/except block).

Then, in `draftjs_exporter/dom.py` we would never import the engines directly (i.e. at the top of the file), but rather use the path import from django (see my comment in `draftjs_exporter/dom.py`) so it always only import the relevant engines and dependencies.

With the current implementation, several things could happen:
- None of the dependencies are available (or the dependencies for the engine selected in the DOM object is missing), the import error will be silenced and an AttributeError will be raise when calling `create_tag` on the selected engine (e.g. for the `DOM_HTML5LIB` engine, when `create_tag` is called, you'll get `AttributeError: dom_engine module has no attribute 'soup'` which will be pretty unhelpful),
- Both `BeautifulSoup` and `lxml` are available in the virtual environment and will both be loaded.

With the proposed solution above, only the selected engine will be loaded and it is missing some dependencies, it will raise an `ImportError` which will make it pretty obvious that you are missing dependencies.